documentapi/src/test/java/com/yahoo/documentapi/messagebus/MessageBusDocumentApiTestCase.java

```diff
@@ -119,17 +120,24 @@ public void requireThatTimeoutWorks() throws InterruptedException {
         AsyncSession session = access().createAsyncSession(new AsyncParameters());
         DocumentType type = access().getDocumentTypeManager().getDocumentType("music");
         Document doc1 = new Document(type, new DocumentId("id:ns:music::1"));
-        assertTrue(session.put(new DocumentPut(doc1),
-                               DocumentOperationParameters.parameters()
-                                                          .withResponseHandler(result -> {
-                                                              response.set(result);
-                                                              latch.countDown();
-                                                          })
-                                                          .withDeadline(Instant.now().minusSeconds(1)))
-                          .isSuccess());
-        assertTrue(latch.await(60, TimeUnit.SECONDS));
-        assertNotNull(response.get());
-        assertEquals(Response.Outcome.TIMEOUT, response.get().outcome());
+
+        // The setup is broken. Zero or negative timeout has semantics in messagebus, so ensuring a timeout is impossible,as negative
+        // remaining time is converted to 1 ms further down. We therefore try tenfold, and accept > 0 timeouts as success ...  (；☉_☉)
+        // We could return a timeout from the destination, but then we're not testing what we want, namely that the sender times out.
```

Comment:
Are there any guarantees that this will trigger any timeouts at all? Looks like the PR build failed with all responses completing successfully. It looks like all messages are routed to an instance of the `Destination` test mock class, so could instead add functionality to it for deferring response sending. The test could then wait for a client-generated timeout response before allowing the server response to be sent. No need for a retry loop either, since it should be deterministic.